File: app/CausalityCSharp/CausalityLibrary/Type/Role.cs

using System;

namespace CausalityLibrary.Type
{
    [Serializable]
    public sealed class Role
    {
        public SerialNumber SerialNumber
        { get; }

        public string Description
        { get; }

        public string Name
        { get; }

        public PositionEnum Position
        { get; }

        public string Voice
        { get; }

        public Role(SerialNumber serialNumber, string description,
            string name, PositionEnum position, string voice)
        {
            SerialNumber = serialNumber;
            Description = description;
            Name = name;
            Position = position;
            Voice = voice;
        }

        // Don't instantiate with default constructor!
        private Role() { }
    }
}

File: app/CausalityJava/src/com/inha/dsp/causality/type/Role.java
package com.inha.dsp.causality.type;

public class Role {
    private SerialNumber serialNumber;
    private String description;
    private String name;
    private PositionEnum position;

    public SerialNumber getSerialNumber() {
        return serialNumber;
    }
    public String getDescription() {
        return description;
    }
    public String getName() {
        return name;
    }
    public PositionEnum getPosition() {
        return position;
    }

    public Role(SerialNumber sn, String desc, String na, PositionEnum pos)
    {
        serialNumber = sn;
        description = desc;
        name = na;
        position = pos;
    }

    // Don't instantiate with default constructor!
    Role() {}
}
